contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.9.0--3.0.0.sql

```diff
@@ -99,5 +99,1292 @@ $$
 STRICT
 LANGUAGE plpgsql IMMUTABLE;
 
+CREATE OR REPLACE FUNCTION sys.babelfish_conv_string_to_date(IN p_datestring TEXT,
```

Comment:
1. Looks like we are re-creating these objects which we permanently dropped in 14.12 which doesn't seem right. I think we should rather have a check before deprecating the object to see if exists. Dropping a non-existing object would be fine since `babelfish_drop_deprecated_object` ignores non-existing functions.
2. These inter-version upgrade scripts are only meant for those objects which needs to updated for the compatibility with the new major version.

Reply:
Here we are re-creating these objects because upgrade script 3.0.0 to 3.1.0 is using these dropped/deprecated objects and will fail if these objects does not exists, causing upgrade to fail.

Reply:
Then let's fix 3.0.0--3.1.0 upgrade script (or any other upgrade script) instead.

Reply:
ok